### ISSUE TITLE
Debug::message -> Logger::debug

### ISSUE THREAD

**Issue by sminnee** [silverstripe/silverstripe-framework#6806]
If we send the Debug::message() content to the PSR-3 logger, then it's much easier to customise where this output is sent, e.g. to [PHP Console](https://chrome.google.com/webstore/detail/php-console/nfhmhhlpfleoednkpnnnkolmclajemef/related?hl=en)

This could be a simple [patch to Debug::message](https://gist.github.com/5316bbd515b126270da4a40203d8b0a3
) however I think we'd want to cover off 2 things:

 * A default implementation that still showed the errors somewhere without a special browser plug-in. The current on-screen display is probably sufficient
 * The other display methods in debug should probably be re-worked to also do this and may need a bit more thought as they produce multi-line output
 * The `!isLive()` check can probably be removed, and instead live environments can have different handling of debug-level messages.

**Comment by Firesphere** [comment]
The quick Gist was a great work, but as we discussed, it will take away realtime logging from Firefox Devs (like me, yea yea :P ), as Firefox doesn't support it yet, and Firephp/FireBug are being phased out and/or abandoned in favor of Firefox Developer Edition.

**Comment by sminnee** [comment]
Yeah we'd need a default option for dev environments to show the errors without a browser plugin. The existing "echo" scheme seems fine for something to bundle with framework

**Comment by Firesphere** [comment]
My personal preference would be to go with a possibly temporary {If !chrome then JS output} system, or even better, have LeKoala's DebugBar ready for SS4 and integrate with that? It already shows messages and such from Debug, why not use it?

**Comment by lekoala** [comment]
Actually, I think the solution could be really simple. Currently, I'm not very happy how I handle the debug messages (I'm buffering everything before real page output, then use ugly regexes to extract debug messages).
So, if it was logging to a logger instead, I could simply use a custom logger that will output everything in the DebugBar. For those who like to see messages displayed on screen, why not have a configuration that allows that instead of dropping the functionality?

I haven't installed SilverStripe 4 yet, but I don't except it's going to be very difficult to adjust my code

**Comment by sminnee** [comment]
I think the cleanest approach, rather than having specific config options, is to handle it all as different monolog log handlers.

So @lekoala updating your module to include a monolog log handler would work well, and it can come bundled with some default config to apply this to the default logger

**Comment by lekoala** [comment]
works for me indeed. and if someone wants the old behaviour, it should be easy to implement a logger that prints everything on the page as it does currently

**Comment by sminnee** [comment]
I'm working on a PR that does that

**Comment by sminnee** [comment]
As long as we retain the functioning of Debug::message() then this can be a minor change, IMO
